docs/concepts/reranking.mdx

```diff
@@ -0,0 +1,64 @@
+---
+title: "Reranking"
+---
+
+Vector search or keyword-based functions (like BM25 or ANN) are optimized for speed, not perfect accuracy.
+They can fetch the right ballpark of results quickly, but the **order might not reflect true relevance**.
+
+Both are designed to efficiently retrieve a shortlist of candidate documents from large collections:
+
+* **BM25**: Uses inverted indexes and term frequencies to quickly score documents based on keyword matches.
+* **ANN (Approximate Nearest Neighbor)**: Uses structures like HNSW, IVF, or PQ to rapidly find vectors that are close to the query vector—but not guaranteed to be the exact top-k closest (hence "approximate").
+
+Reranking is used to improve relevance and accuracy. TopK comes with built in-built reranking functionality.
+
+## `rerank()` function
+
+You can rerank the results of a query using the `rerank()` function which accepts the following parameters:
+
+<ParamField query="model" type="string" default="cohere/rerank-v3.5">
+  The model to use for reranking. Currently, only [`cohere/rerank-v3.5`](https://cohere.com/blog/rerank-3pt5) is supported.
+</ParamField>
+
+<ParamField query="query" type="string" optional>
+  The query text to rerank against. Uses arguments from `semantic_similarity()` function if not specified.
+</ParamField>
+
+<ParamField query="fields" type="string[]" optional>
```

Comment:
I think this only works if you have a single `semantic_similarity()` call. if you have multiple, you must provide the query. but we should check, not sure